### PR TITLE
[SPARK-45447][MLLIB][TESTS] Reduce the memory required to start the `LocalClusterSparkContext` in the `mllib` module test cases.

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
This pr aims to reduce the chance of GA test tasks being killed by decreasing the memory required to start `LocalClusterSparkContext`.

### Why are the changes needed?
The GA test task for mllib is sometimes killed, and the test cases to be tested that often get killed are those that inherit from `LocalClusterSparkContext`, such as

- LBFGSClusterSuite: https://github.com/apache/spark/actions/runs/6421930285/job/17437383223
- NaiveBayesClusterSuite: https://github.com/apache/spark/actions/runs/6436807001/job/17480899578
- LassoClusterSuite: https://github.com/apache/spark/actions/runs/6362618008/job/17289887813
- LBFGSClusterSuite: https://github.com/apache/spark/actions/runs/6346972272/job/17289897396

So this pr circumvents this issue by reducing the startup memory required for `LocalClusterSparkContext`.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Should monitor GA

### Was this patch authored or co-authored using generative AI tooling?
No

### CODE REVIEW

**Comment by LuciferYang** [comment]
Monitor whether the GA test task `Run / Build modules: streaming, sql-kafka-0-10, streaming-kafka-0-10, mllib-local, mllib, yarn, kubernetes, hadoop-cloud, spark-ganglia-lgpl, connect, protobuf` will be more stable.

**Comment by srowen** [comment]
Looks OK if it passes

**Comment by LuciferYang** [comment]
> Looks OK if it passes

All passed

File: mllib/src/test/scala/org/apache/spark/mllib/util/LocalClusterSparkContext.scala [hunk 1]
```diff
@@ -28,8 +29,9 @@ trait LocalClusterSparkContext extends BeforeAndAfterAll { self: Suite =>
   override def beforeAll(): Unit = {
     super.beforeAll()
     val conf = new SparkConf()
-      .setMaster("local-cluster[2, 1, 1024]")
+      .setMaster("local-cluster[2, 1, 512]")
       .setAppName("test-cluster")
+      .set(EXECUTOR_MEMORY.key, "512m")
```
Comment (LuciferYang) [inline mllib/src/test/scala/org/apache/spark/mllib/util/LocalClusterSparkContext.scala:34]: change to use `ConfigEntry`, test it one more time.

**Comment by LuciferYang** [comment]
All test passed

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @LuciferYang .
Merged to master.

**Comment by LuciferYang** [comment]
Thanks @dongjoon-hyun

**Comment by dongjoon-hyun** [comment]
Unfortunately, the pipeline seems to be still unstable and took over 2h 18minutes sometime.
- https://github.com/apache/spark/actions/runs/6555720246/job/17804607728
- https://github.com/apache/spark/actions/runs/6555731945/job/17804636584

**Comment by LuciferYang** [comment]
> Unfortunately, the pipeline seems to be still unstable and took over 2h 18minutes sometime.
> 
> * https://github.com/apache/spark/actions/runs/6555720246/job/17804607728
> * https://github.com/apache/spark/actions/runs/6555731945/job/17804636584

Yes, I am trying to put `mllib` into the same group as `catalyst,hive-thriftserver` to see if it can stabilize the tests.

**Comment by LuciferYang** [comment]
Additionally, in local testing, I found that the `sbt-launcher` process starts with `-Xmx4g -Xms4g`. This means that for LocalCluster testing, at least three processes will start:sbt-launcher, forkMain and local-cluster, but I have not yet found the correct place to change the `-Xmx` and `-Xms` of `sbt-launcher`.

**Comment by dongjoon-hyun** [comment]
`catalyst, hive-thriftserver` is also over 1h 37min.
- https://github.com/apache/spark/actions/runs/6553891212/job/17799995346

Since `mllib` and  `mllib-local` also needs over 30 minutes, it's hard to put there.

**Comment by LuciferYang** [comment]
Or put them into a separate group?

**Comment by dongjoon-hyun** [comment]
Let's revisit this later. Currently, I'm trying to remove UNIDOC and MIMA from `build` job.

- #43422

Hopefully, minimizing the jobs will give us more options including these mllib module issues.

**Pull request closed without merging**